### PR TITLE
Code Sharing -- Issue #443

### PR DESCRIPTION
**Pull request by rpetrusha**

### CODE REVIEW

File: docs/concepts/codesharing.md [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+
+Portable Class Libraries were intended to provide cross-platform binary libraries, but the APIs available for a particular profile (a particular set of target platforms) was the intersection of the APIs available on each platform. This made it very difficult to determine in advance which APIs were and were not supported. Commonly used APIs were sometimes not present in a particular profile, and it was often not clear why particular APIs had apparently "disappeared."
+
+## The .NET Standard Library and Source Code Compatibility ##
+
+The .NET Standard Library addresses the problem of source code compatibility by defining the API surface area that a particular version of the standard must implement. This surface area is defined by the types and type members found in the reference assemblies that are included in a particular version of the the .NET Standard Library.
```
Comment (mairaw) [inline docs/concepts/codesharing.md]: remove duplicate the - "the the .NET Standard"

File: docs/concepts/codesharing.md [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+
+The .NET Standard Library aims at maintaining binary compatibility between the .NET-capable platforms for which an application is developed and future versions of those platforms. It also supports source code compatibility between platforms that support a particular version of the .NET Standard. 
+
+<aside style="border-style:double">
+**Important:** This is very preliminary documentation for the .NET Standard Library. Because the standard has not been formalized yet, it is subject to change.
+</aside>
+
+The .NET Standard Library consists of two parts: a set of contracts (or reference assemblies) that are included in the standard, and a single moniker-based versioning scheme.
+
+# What Is the .NET Standard Library? #
```
Comment (mairaw) [inline docs/concepts/codesharing.md]: shouldn't this be a sub-header? 
\##  What Is the .NET Standard Library?

**Comment by richlander** [comment]
This has a lot of overlap with the .NET Standard Library doc. We should focus all of the definitional content in that doc. I skipped some of that material (like that image) because I wasn't keen on it. 

I was proposing that you focus this doc exclusively on binary and source sharing and not specific to .NET Standard, but inclusive of it.

File: docs/concepts/codesharing.md [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+      }
+
+      [System.AttributeUsageAttribute((System.AttributeTargets)(1), AllowMultiple = false)]
+      [System.ComponentModel.EditorBrowsableAttribute((System.ComponentModel.EditorBrowsableState)(1))]
+      public sealed partial class SatelliteContractVersionAttribute : System.Attribute
+      {
+         public SatelliteContractVersionAttribute(string version) { }
+         public string Version { get { return default(string); } }
+      }
+   }
```
Comment (svick) [inline docs/concepts/codesharing.md]: Misaligned `}`. (Needs one more space.)

**Comment by rpetrusha** [comment]
Closing to open a new pull request from my own fork.

**Pull request closed without merging**